gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java

```diff
@@ -665,6 +679,7 @@ private void viewFile(ProjectFile file, ProjectFileViewerExtension viewerExtensi
         Node graphic = viewerExtension.getMenuGraphic(file);
         ProjectFileViewer viewer = viewerExtension.newViewer(file, getContent().getScene(), context);
         Tab tab = new MyTab(tabName, viewer);
+        viewerList.add(viewer);
```

Comment:
You never remove the viewer from the list so when the tab is closed and the last action was to discard changes (leave with "not saving" option), its last expectingly discarded changes are saved even if the tab is closed.